week2/week2-HW/index.js

```diff
@@ -0,0 +1,279 @@
+const selectLi = document.getElementById("select-li");
+const selectToggle = document.querySelector(".select-toggle");
+
+selectLi.addEventListener("click", () => {
+  // select-toggle을 보이게 함
+  selectToggle.style.display =
+    selectToggle.style.display === "block" ? "none" : "block";
+});
+
+// 중요도 버튼 클릭 시 필터링
+const selectToggleButtons = document.querySelectorAll(".select-toggle div");
+
+selectToggleButtons.forEach((button) => {
+  button.addEventListener("click", () => {
+    selectToggle.style.display = "none !important"; // 선택 후 toggle 숨기기
+    const priority = parseInt(button.id.replace("select-toggle-", "")); // button의 id에서 중요도 추출
+    const filteredTodos = todos.filter((todo) => todo.priority === priority); // 중요도로 필터링
+    renderTodos(filteredTodos); // 필터링된 todos 렌더링
+  });
+});
+
+// nav 리스트에 클릭 이벤트 추가
+allLi.addEventListener("click", () => {
+  renderTodos(todos); // 전체 목록 렌더링
+});
+
+completeLi.addEventListener("click", () => {
+  const filteredTodos = todos.filter((todo) => todo.completed); // 완료된 항목만 필터링
+  renderTodos(filteredTodos);
+});
+
+incompleteLi.addEventListener("click", () => {
+  const filteredTodos = todos.filter((todo) => !todo.completed); // 완료된 항목만 필터링
+  renderTodos(filteredTodos);
+});
+
+// 일정 데이터 fetch하기!
+function renderTodos(filteredTodos = todos) {
+  // 필터링된 todos 배열이 없을 경우 기본값으로 todos를 사용
+  if (!Array.isArray(filteredTodos)) {
+    console.error("filteredTodos is not an array:", filteredTodos);
+    return;
+  }
+
+  const tbody = document.querySelector("table tbody");
+  tbody.innerHTML = ""; // 기존 내용을 비우고 새로 렌더링
+
+  filteredTodos.forEach((todo) => {
+    const tr = document.createElement("tr");
+    tr.classList.add("todo-data");
+    tr.dataset.id = todo.id;
+
+    const checkboxTd = document.createElement("td");
+    const checkbox = document.createElement("input");
+    checkbox.type = "checkbox";
+    checkboxTd.appendChild(checkbox);
+
+    const priorityTd = document.createElement("td");
+    priorityTd.textContent = todo.priority;
+
+    const completedTd = document.createElement("td");
+    completedTd.textContent = todo.completed ? "✅" : "❌";
+
+    const titleTd = document.createElement("td");
+    titleTd.textContent = todo.title;
+
+    tr.appendChild(checkboxTd);
+    tr.appendChild(priorityTd);
+    tr.appendChild(completedTd);
+    tr.appendChild(titleTd);
+
+    tbody.appendChild(tr);
+  });
+}
+
+// 체크박스 제어 (all-check 쪽)
+let isAll = false;
+const controlCheckBtn = document.querySelector("table thead #all-check");
+const checkboxes = document.querySelectorAll(
+  'table tbody input[type="checkbox"]'
+);
+// 전체 체크박스임
+controlCheckBtn.addEventListener("click", function () {
+  const checkbox = document.querySelectorAll(
+    'table tbody input[type="checkbox"]'
+  );
+  if (isAll == false) {
+    checkbox.forEach((cb) => {
+      cb.checked = true;
+    });
+  } else {
+    checkbox.forEach((cb) => {
+      cb.checked = false;
+    });
+  }
+  // isAll 값 토글
+  isAll = !isAll;
```

Comment:
```suggestion
  isAll = !isAll;
  checkbox.forEach((cb) => {
    cb.checked = isAll;
  });
```
`isAll`의 값을 반전시켜야 한다는 점을 이용해서 조금 더 간결하게 나타낼 수 있을 것 같아요!